lib/pure/asyncftpclient.nim

```diff
@@ -368,8 +409,7 @@ proc doUpload(ftp: AsyncFtpClient, file: File,
   while ftp.dsockConnected:
     if sendFut == nil or sendFut.finished:
       # TODO: Async file reading.
-      let len = file.readBuffer(addr(data[0]), 4000)
-      setLen(data, len)
```

Comment:
Why removing this? If we don't do this I believe we might read uninitialised bytes.

Reply:
solved https://github.com/nim-lang/Nim/pull/17219/commits/7d49cc7feb7642e48ae544600fcd044ba64d2b74